datamodel/initdb.d/05_reference_implementation_support.sql

```diff
@@ -147,16 +116,47 @@ UNION
     operations_event.port_call_service_type_code,
     operations_event.facility_type_code,
     operations_event.vessel_position,
-    operations_event.publisher,
-    (
-        SELECT DISTINCT s.carrier_booking_reference
-        FROM dcsa_im_v3_0.shipment s
-        JOIN dcsa_im_v3_0.shipment_transport st ON s.id = st.shipment_id
-        JOIN dcsa_im_v3_0.transport t ON st.transport_id = t.id
-        WHERE t.discharge_transport_call_id = transport_call_id
-    ) AS carrier_booking_reference
+    operations_event.publisher
    FROM dcsa_im_v3_0.operations_event;
 
+DROP VIEW IF EXISTS dcsa_im_v3_0.event_carrier_booking_reference CASCADE;
+CREATE VIEW dcsa_im_v3_0.event_carrier_booking_reference AS
+    SELECT DISTINCT s.carrier_booking_reference,
+                    'TC_ID' AS link_type,
+                    COALESCE(t.load_transport_call_id, t.discharge_transport_call_id) AS transport_call_id,
+                    null AS "document_id"
+    FROM dcsa_im_v3_0.shipment s
+    JOIN dcsa_im_v3_0.shipment_transport st ON s.id = st.shipment_id
+    JOIN dcsa_im_v3_0.transport t ON st.transport_id = t.id
+   UNION
+    SELECT DISTINCT s.carrier_booking_reference,
+                    'TRD' AS link_type,
+                    null AS transport_call_id,
+                    -- Should be transport document ID when we are getting document versioning.
+                    td.transport_document_reference AS document_id
+    FROM dcsa_im_v3_0.transport_document td
+    JOIN dcsa_im_v3_0.cargo_item ci ON td.shipping_instruction_id = ci.shipping_instruction_id
+    JOIN dcsa_im_v3_0.shipment_equipment se ON se.id = ci.shipment_equipment_id
+    JOIN dcsa_im_v3_0.shipment s ON se.shipment_id = s.id
+   UNION
+    SELECT DISTINCT s.carrier_booking_reference,
+                    'SHI' AS link_type,
+                    null AS transport_call_id,
+                    -- Should be shipping instruction ID when we are getting document versioning.

```

Comment:
I assume this needs to be renamed as part of the PR @douradofrazer is creating (renaming SI_ID --> SIReference) - just so we don't forget...

Reply:
This is exactly why I added the comment because I knew it was coming. :)